### PR TITLE
create spy res to capture response from hosts and applications calls …

### PR DESCRIPTION
**Pull request by nicktate**
…so we do not hit race condition overwriting the same res object

@normanjoyner @ashleyschuett

### CODE REVIEW

**Comment by ashleyschuett** [comment]
👍

File: handlers/v1/cluster.js [hunk 1]
```diff
@@ -40,21 +40,23 @@ module.exports = {
...
 
-                    return fn(null, res.stash.body);
+                    return fn(null, spy.stash.body);
                 });
             },
             hosts: (fn) => {
-                hosts.get(req, res, () => {
+                const spy = { stash: {} };
+                hosts.get(req, spy, () => {
                     if(res.stash.code !== 200) {
```
Comment (ashleyschuett) [inline handlers/v1/cluster.js]: `spy.stash.code !== 200`

File: handlers/v1/cluster.js [hunk 2]
```diff
@@ -40,21 +40,23 @@ module.exports = {
     state: function(req, res, next) {
         async.parallel({
             applications: (fn) => {
-                applications.get(req, res, () => {
+                const spy = { stash: {} };
+                applications.get(req, spy, () => {
                     if(res.stash.code !== 200) {
```
Comment (ashleyschuett) [inline handlers/v1/cluster.js]: `spy.stash.code !== 200`

**Comment by nicktate** [comment]
good catch

**Pull request merged**